
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="..">
      
      
        <link rel="next" href="../comparing-metagenomes/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.20">
    
    
      
        <title>Analyzing a single metagenome for taxonomy - Metagenomics @ 2024 PIG-PARADIGM annual meeting</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.66ac8b77.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="ucdavis" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#analyzing-a-single-metagenome-for-taxonomy" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Metagenomics @ 2024 PIG-PARADIGM annual meeting" class="md-header__button md-logo" aria-label="Metagenomics @ 2024 PIG-PARADIGM annual meeting" data-md-component="logo">
      
  <img src="../assets/ucdavis_logo_gold.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics @ 2024 PIG-PARADIGM annual meeting
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Analyzing a single metagenome for taxonomy
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="ucdavis" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/ngs-docs/2024-pig-paradigm-workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Metagenomics @ 2024 PIG-PARADIGM annual meeting" class="md-nav__button md-logo" aria-label="Metagenomics @ 2024 PIG-PARADIGM annual meeting" data-md-component="logo">
      
  <img src="../assets/ucdavis_logo_gold.png" alt="logo">

    </a>
    Metagenomics @ 2024 PIG-PARADIGM annual meeting
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ngs-docs/2024-pig-paradigm-workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
      
        
          
        
      
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1" >
        
          
          
          <div class="md-nav__link md-nav__container">
            <a href=".." class="md-nav__link ">
              
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

            </a>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Introduction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Tutorials
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Analyzing a single metagenome for taxonomy
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Analyzing a single metagenome for taxonomy
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#creating-a-working-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Creating a working directory
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#installation" class="md-nav__link">
    <span class="md-ellipsis">
      Installation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#basic-membership-analysis-of-the-metagenome" class="md-nav__link">
    <span class="md-ellipsis">
      Basic membership analysis of the metagenome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Basic membership analysis of the metagenome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#k-mer-abundance-histogram" class="md-nav__link">
    <span class="md-ellipsis">
      K-mer abundance histogram
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#k-mer-overlap-between-metagenomes-and-two-genomes" class="md-nav__link">
    <span class="md-ellipsis">
      K-mer overlap between metagenomes and two genomes
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summarizing-taxonomy" class="md-nav__link">
    <span class="md-ellipsis">
      Summarizing taxonomy
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comparing-metagenomes/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Comparing metagenomes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../amr/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Analyzing metagenomes for Antimicrobial Resistance (AMR) Genes
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#creating-a-working-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Creating a working directory
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#installation" class="md-nav__link">
    <span class="md-ellipsis">
      Installation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#basic-membership-analysis-of-the-metagenome" class="md-nav__link">
    <span class="md-ellipsis">
      Basic membership analysis of the metagenome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Basic membership analysis of the metagenome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#k-mer-abundance-histogram" class="md-nav__link">
    <span class="md-ellipsis">
      K-mer abundance histogram
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#k-mer-overlap-between-metagenomes-and-two-genomes" class="md-nav__link">
    <span class="md-ellipsis">
      K-mer overlap between metagenomes and two genomes
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summarizing-taxonomy" class="md-nav__link">
    <span class="md-ellipsis">
      Summarizing taxonomy
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="analyzing-a-single-metagenome-for-taxonomy">Analyzing a single metagenome for taxonomy<a class="headerlink" href="#analyzing-a-single-metagenome-for-taxonomy" title="Permanent link">&para;</a></h1>
<p>The tutorial uses <a href="https://sourmash.readthedocs.io/">sourmash</a> to do
various k-mer based analyses of Illumina shotgun metagenome content.</p>
<p>In this tutorial, you will learn:</p>
<ul>
<li>how to look at what genomes share content with a metagenome;</li>
<li>how to examine the abundance of metagenome content without a reference;</li>
<li>how to summarize the taxonomic content of a metagenome;</li>
</ul>
<p>We will be using the taxonomic classification system as benchmarked in
<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-022-05103-0">Evaluation of taxonomic classification and profiling methods for long-read shotgun metagenomic sequencing datasets</a>,
which is both very <em>sensitive</em> and quite <em>specific</em>.</p>
<h2 id="creating-a-working-directory">Creating a working directory<a class="headerlink" href="#creating-a-working-directory" title="Permanent link">&para;</a></h2>
<p>Run:
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>mkdir ~/single-metag/
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>cd ~/single-metag/
</code></pre></div></p>
<h2 id="installation">Installation<a class="headerlink" href="#installation" title="Permanent link">&para;</a></h2>
<p>First let's install the necessary software. We'll be using
<a href="https://sourmash.readthedocs.io/en/latest/index.html">the sourmash software</a>
along with a few sourmash plugins.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>mamba create -n tax -y sourmash sourmash_plugin_branchwater
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>conda activate tax
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>pip install sourmash_plugin_abundhist sourmash_plugin_venn
</code></pre></div>
<h2 id="basic-membership-analysis-of-the-metagenome">Basic membership analysis of the metagenome<a class="headerlink" href="#basic-membership-analysis-of-the-metagenome" title="Permanent link">&para;</a></h2>
<p>Now let's convert the CD136 metagenome into sourmash sketch format
<a href="https://sourmash.readthedocs.io/en/latest/command-line.html#sourmash-sketch-make-sourmash-signatures-from-sequence-data">(docs)</a>:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>sourmash sketch dna -p abund \
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>    ../data/IBD_tutorial_subset/metag/1-trimmed/CD136/*.fq.gz \
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>    --name CD136 -o CD136.sig.zip
</code></pre></div>
<p>And search it against the latest
<a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a>
containing all known genomes. We'll do the primary search with the
'fastgather' extension to sourmash.</p>
<p>Run:
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>sourmash scripts fastgather CD136.sig.zip \
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>    ../databases/gtdb-rs214-k31.zip -o CD136.x.gtdb-rs214.fastgather.csv -c 16
</code></pre></div>
(This will take about 3 minutes.)</p>
<p>Display the results of the first search in a nicer format using
'sourmash gather'
<a href="https://sourmash.readthedocs.io/en/latest/classifying-signatures.html#analyzing-metagenomic-samples-with-gather">(docs</a>.</p>
<p>Run:
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>sourmash gather CD136.sig.zip \
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>    ../databases/gtdb-rs214-k31.zip -o CD136.x.gtdb-rs214.gather.csv \
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>    --picklist CD136.x.gtdb-rs214.fastgather.csv:match_name:ident
</code></pre></div></p>
<p>You should see the output below. What does this mean?
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>overlap     p_query p_match avg_abund
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>---------   ------- ------- ---------
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>5.0 Mbp       27.5%   96.7%       7.3    GCF_000598785.2 Bacteroides fragilis...
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>3.6 Mbp       10.3%   64.2%       3.8    GCF_009678525.1 Parabacteroides dist...
<a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>3.2 Mbp        4.6%   59.0%       1.9    GCF_015550345.1 Bacteroides uniformi...
<a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>1.6 Mbp       30.7%   58.4%      25.3    GCA_023708525.1 Faecalibacterium pra...
<a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>3.6 Mbp        1.2%    8.0%       3.8    GCF_009024595.1 Parabacteroides dist...
<a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a>1.6 Mbp        5.6%   10.2%      25.1    GCF_017377615.1 Faecalibacterium sp....
<a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>2.6 Mbp        0.5%    3.3%       3.7    GCF_015548395.1 Parabacteroides dist...
<a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a>1.6 Mbp        2.3%    5.1%      24.6    GCA_905199165.1 Faecalibacterium pra...
<a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a>3.5 Mbp        0.2%    1.9%       2.8    GCA_009678725.1 Parabacteroides dist...
<a id="__codelineno-5-12" name="__codelineno-5-12" href="#__codelineno-5-12"></a>2.1 Mbp        0.1%    1.5%       1.9    GCF_009020375.1 Bacteroides uniformi...
<a id="__codelineno-5-13" name="__codelineno-5-13" href="#__codelineno-5-13"></a>3.3 Mbp        0.2%    1.5%       3.6    GCF_015552355.1 Parabacteroides dist...
<a id="__codelineno-5-14" name="__codelineno-5-14" href="#__codelineno-5-14"></a>1.6 Mbp        1.4%    2.4%      25.2    GCF_000166035.1 Faecalibacterium cf....
<a id="__codelineno-5-15" name="__codelineno-5-15" href="#__codelineno-5-15"></a>4.0 Mbp        0.3%    1.1%       7.3    GCF_009024655.1 Bacteroides fragilis...
<a id="__codelineno-5-16" name="__codelineno-5-16" href="#__codelineno-5-16"></a>1.6 Mbp        1.0%    1.9%      25.3    GCA_019425405.1 Faecalibacterium sp....
<a id="__codelineno-5-17" name="__codelineno-5-17" href="#__codelineno-5-17"></a>found less than 50.0 kbp in common. =&gt; exiting
<a id="__codelineno-5-18" name="__codelineno-5-18" href="#__codelineno-5-18"></a>
<a id="__codelineno-5-19" name="__codelineno-5-19" href="#__codelineno-5-19"></a>found 14 matches total;
<a id="__codelineno-5-20" name="__codelineno-5-20" href="#__codelineno-5-20"></a>the recovered matches hit 86.0% of the abundance-weighted query.
<a id="__codelineno-5-21" name="__codelineno-5-21" href="#__codelineno-5-21"></a>the recovered matches hit 61.7% of the query k-mers (unweighted).
</code></pre></div></p>
<p>Points to discuss:</p>
<ul>
<li>13 matches were found! This means that significant portions of 13
  different GenBank genomes matched to some part of the metagenome.</li>
<li>based on the abundance-weighted matches, 86.0% of the metagenome
  reads will map to at least one of these genomes. This means that
  at least 86% of the metagenome content is known!</li>
<li>based on the unweighted matches, 61.7% of the metagenome's genomic
  content is present in the reference database.  Some of this is
  probably erroneous data or host contamination.</li>
</ul>
<h3 id="k-mer-abundance-histogram">K-mer abundance histogram<a class="headerlink" href="#k-mer-abundance-histogram" title="Permanent link">&para;</a></h3>
<p>Let's examine this data set further. First, let's take a look at the
abundance spectrum of the metagenome content. This is an annotation and
reference-free exercise: we're just counting content.</p>
<p>Run the abundhist plugin:
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>sourmash scripts abundhist CD136.sig.zip \
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>    --max 50 --bins 20 --figure CD136-abundance.png \
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>    --ymax 4000
</code></pre></div>
This will produce a file <code>CD136-abundance.png</code>, which will look approximately
like this:</p>
<p><img alt="Abundance histogram of content in CD136" src="../images/CD136-abundance.png" />)</p>
<p>Points to discuss:</p>
<ul>
<li>this is a histogram of sequence content, separated by abundance.</li>
<li>the X axis is the abundance, the Y axis is how much of the content is at
  that abundance.</li>
<li>you would expect a peak for each species present.</li>
<li>for this simple mock metagenome, there are two peaks: one under 10, one around 30.</li>
<li>we do not know what these species are based on this diagram, but we can annotate them based on the avg_abund column in the sourmash gather output above.</li>
</ul>
<h3 id="k-mer-overlap-between-metagenomes-and-two-genomes">K-mer overlap between metagenomes and two genomes<a class="headerlink" href="#k-mer-overlap-between-metagenomes-and-two-genomes" title="Permanent link">&para;</a></h3>
<p>Now, since (based on the gather output and the abundance histogram) we
think there are two major species, let's take a look at the overlap of
genomes of these species with the entire metagenome.</p>
<p>We do this with the Venn plugin. But first we need to extract and rename
the matches we're interested in comparing:</p>
<p>Run:
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>sourmash sig grep &quot;GCF_000598785.2&quot; ../databases/gtdb-rs214-k31.zip \
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>    -o GCF_000598785.2.sig.zip
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>sourmash sig grep &quot;GCA_023708525.1&quot; ../databases/gtdb-rs214-k31.zip \
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>    -o GCA_023708525.1.sig.zip
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>
<a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a>sourmash sig rename GCF_000598785.2.sig.zip &quot;B. fragilis I1345&quot; \
<a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a>    -o B.fragilis.sig.zip
<a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a>sourmash sig rename GCA_023708525.1.sig.zip &quot;F. prausnitzii&quot; \
<a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a>    -o F.prausnitzii.sig.zip
</code></pre></div></p>
<p>And now generate a 3-way Venn diagram:
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>sourmash scripts venn -k 31 CD136.sig.zip B.fragilis.sig.zip \
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>    F.prausnitzii.sig.zip -o CD136-venn.png
</code></pre></div></p>
<p>In the resulting figure <code>CD136-venn.png</code> (see below), multiply numbers
by 1000 to get number of bases. CD136 (the red circle) is the entire
metagenome.</p>
<p><img alt="Venn diagram of top two species matches" src="../images/CD136-venn.png" /></p>
<p>Points to discuss:</p>
<ul>
<li>the metagenome CD136 contains a mixture of content from whatever
  genomes are present.</li>
<li>based on the gather output, we can see two major species (also see
  taxonomy results below): <em>B. fragilis</em> and <em>F. prausnitzii</em>.</li>
<li>This Venn diagram illustrates their genomic overlap with the k-mer
  content of the metagenome. This is not weighted by abundance: the size
  of the circles is relative to the genome size.</li>
<li>We can see that the <em>B. fragilis I135</em> genome is almost entirely
  contained in the metagenome. Based on the p_match column from the gather
  output above, it is 96.7% present in the metagenome!</li>
<li>In contrast, the <em>F. prausnitzii</em> genome is just a bit more than half
  contained in the metagenome (p_match is </li>
</ul>
<p>Questions to ask:</p>
<ul>
<li>why are these genomes not entirely present (100% overlap)?<ul>
<li>low coverage</li>
<li>strain variation</li>
</ul>
</li>
<li>what ELSE might be in the metagenome??<ul>
<li>contamination</li>
<li>unknown genomes (see morning session)</li>
</ul>
</li>
</ul>
<h2 id="summarizing-taxonomy">Summarizing taxonomy<a class="headerlink" href="#summarizing-taxonomy" title="Permanent link">&para;</a></h2>
<p>Prepare the taxonomic lineages file so that the commands run fast.</p>
<p>Run:
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>sourmash tax prepare -t ../databases/gtdb-rs214.lineages.csv.gz \
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>    -F sql -o gtdb-rs214.lineages.sqldb
</code></pre></div></p>
<p>Now let's generate a human-readable taxonomic analysis at the species
level. This uses the GTDB taxonomic lineages to summarize all of the
gather matches at the species level.</p>
<p>Run:
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>sourmash tax metagenome -g CD136.x.gtdb-rs214.gather.csv \
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>    -t gtdb-rs214.lineages.sqldb -F human
</code></pre></div></p>
<p>You will see:
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>sample name    proportion   cANI   lineage
<a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>-----------    ----------   ----   -------
<a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>CD136             41.0%     92.5%  d__Bacteria;p__Bacillota_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Faecalibacterium;s__Faecalibacterium prausnitzii_D
<a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>CD136             27.8%     95.1%  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides fragilis
<a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>CD136             14.1%     -      unclassified
<a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a>CD136             12.4%     94.7%  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis
<a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a>CD136              4.6%     93.7%  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis
</code></pre></div></p>
<p>we can also roll this up from species to other taxonomic ranks, like
class, by adding <code>-r &lt;rank&gt;</code>.</p>
<p>Run:
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>sourmash tax metagenome -g CD136.x.gtdb-rs214.gather.csv \
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>    -t gtdb-rs214.lineages.sqldb  -F human -r class
</code></pre></div></p>
<p>and you will get the above results "pulled back" to class:
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>sample name    proportion   cANI   lineage
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>-----------    ----------   ----   -------
<a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a>CD136             44.9%     98.0%  d__Bacteria;p__Bacteroidota;c__Bacteroidia
<a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a>CD136             41.0%     92.5%  d__Bacteria;p__Bacillota_A;c__Clostridia
<a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a>CD136             14.1%     -      unclassified
</code></pre></div></p>
<p>Points to discuss:</p>
<ul>
<li>what is the "unclassified" content?</li>
<li>what is cANI? (what is ANI!)</li>
</ul>
<p><strong>Things for students to try out:</strong></p>
<ul>
<li>Try executing the commands for different ranks - genus, family,
  order, class, phylum, superkingdom. Do the results make sense?</li>
</ul>
<hr />
<p>Next tutorial: <a href="../comparing-metagenomes/">Comparing metagenomes</a></p>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1-2.1-2M12.5 7v5.2l4 2.4-1 1L11 13V7h1.5M11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2v1.8Z"/></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">April 29, 2024</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3h-2Z"/></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">April 29, 2024</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href=".." class="md-footer__link md-footer__link--prev" aria-label="Previous: Introduction">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Introduction
              </div>
            </div>
          </a>
        
        
          
          <a href="../comparing-metagenomes/" class="md-footer__link md-footer__link--next" aria-label="Next: Comparing metagenomes">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Comparing metagenomes
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023-2024 Titus Brown and Anneliek ter Horst
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.code.annotate", "navigation.sections", "navigation.expand", "navigation.indexes", "navigation.top", "navigation.footer", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.dd8806f2.min.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>